In pycairo-1.8.4 this takes two args, but that version requires python2.6.

--- origsrc/pygoocanvas-0.14.1/goocanvasmodule.c	2009-05-09 05:01:38.000000000 -0500
+++ src/pygoocanvas-0.14.1/goocanvasmodule.c	2009-06-09 21:36:38.228975700 -0500
@@ -35,7 +35,7 @@ _cairo_matrix_to_gvalue(GValue *value, P
 static PyObject *
 _cairo_pattern_from_gvalue(const GValue *value)
 {
-    return PycairoPattern_FromPattern(cairo_pattern_reference((cairo_pattern_t *) g_value_get_boxed(value)), NULL);
+    return PycairoPattern_FromPattern(cairo_pattern_reference((cairo_pattern_t *) g_value_get_boxed(value)));
 }
 
 static int
